pynumdiff/utils/evaluate.py

```diff
@@ -118,3 +118,19 @@ def error_correlation(dxdt_hat, dxdt_truth, padding=None):
     r = stats.linregress(dxdt_truth[padding:-padding] -
                                 np.mean(dxdt_truth[padding:-padding]), errors)
     return r.rvalue**2
+
+
+def total_variation(x, padding=0):
+    """Calculate the total variation of an array. Used by optimizer.
+
+    :param np.array[float] x: data
+
+    :return: (float) -- total variation
+    """
+    if np.isnan(x).any():
+        return np.nan
+    if padding is None or padding == 'auto':
+        padding = int(0.025*len(x))
+        padding = max(padding, 1)
+    
+    return np.sum(np.abs(x[1:]-x[:-1]))/(len(x)-1)  # mostly equivalent to cvxpy.tv(x2-x1).value
```

Comment:
"mostly", yeah whatever is going on in this code is slightly difficult to follow: https://www.cvxpy.org/_modules/cvxpy/atoms/total_variation.html#tv